src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaElementDecl.cs

```diff
@@ -90,10 +91,10 @@ IDtdAttributeInfo IDtdAttributeListInfo.LookupAttribute(string prefix, string lo
 
         IEnumerable<IDtdDefaultAttributeInfo> IDtdAttributeListInfo.LookupDefaultAttributes()
         {
-            return _defaultAttdefs;
+            return _defaultAttdefs!;
```

Comment:
I suspect this is unexpected here and will cause some kind of error but given that interface mentions null on all methods returning null but it does not on this one:

```
        /// <summary>
        /// Returns enumeration of all default attributes
        /// defined in this attribute list.
        /// </summary>
        /// <returns>Enumerator of default attribute.</returns>
```

Also I suspect this will never happen

Reply:
OK

Reply:
There's also one null check for this method on https://github.com/dotnet/runtime/blob/6904681fb5697febccd1e37d74e6292792dc4b8e/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs#L4658-L4660

Is it really possible for this to never return null? I would suggest to remove the check and see if any test fails, if it does, we should probably add `?` here.

Reply:
I probably won't have time to investigate this - honestly I'd refrain from removing checks in nullability PR unless you're 100% sure of the change

Reply:
> Is it really possible for this to never return null? 

It is possible to return null if the `_defaultAttdefs` is not set before calling this method, that is why I commented here, but @krwq replied basically even from the code it can be null that is not the intention, I accept his point and as it is not public API I am fine with suppressing it.

> I would suggest to remove the check and see if any test fails, if it does, we should probably add ? here and on base type IDtdAttributeListInfo.
> If nothing fails, then it would be good to remove said null check on XmlTextReaderImpl.

Removing null check for such case is not suggested, I would remove `!` and return `?` instead. 

@Jozkee as the annotations in this PR needed for further annotations I want to merge it ASAP, if you still want to make it return nullable let me now, I can address this comment as @krwq is OOO.

Reply:
Talked with @Jozkee offline, he is OK with merging, he will update it if he sees that is needed with his future nullability PR